src/app/modules/main/profile_section/wallet/accounts/module.nim

```diff
@@ -0,0 +1,131 @@
+import tables, NimQml, sequtils, sugar, chronicles
+
+import ./io_interface, ./view, ./item, ./controller, ./utils
+import ../io_interface as delegate_interface
+import ../../../../../global/global_singleton
+import ../../../../../core/eventemitter
+import ../../../../../../app_service/common/account_constants
+import ../../../../../../app_service/service/keycard/service as keycard_service
+import ../../../../../../app_service/service/wallet_account/service as wallet_account_service
+import ../../../../../../app_service/service/network/service as network_service
+import ../../../../shared_modules/keycard_popup/io_interface as keycard_shared_module
+
+export io_interface
+
+# TODO: remove it completely if after wallet settings part refactore this is not needed.
+type
+  AuthenticationReason {.pure.} = enum
+    DeleteAccountAuthentication = 0
+
+# TODO: remove it completely if after wallet settings part refactore this is not needed.
+type WalletAccountDetails = object
+  address: string
+  path: string
+  addressAccountIsDerivedFrom: string
+  publicKey: string
+  keyUid: string
+
+type
+  Module* = ref object of io_interface.AccessInterface
+    delegate: delegate_interface.AccessInterface
+    events: EventEmitter
+    view: View
+    controller: Controller
+    moduleLoaded: bool
+    walletAccountService: wallet_account_service.Service
+    processingWalletAccount: WalletAccountDetails
+    authentiactionReason: AuthenticationReason
+
+proc newModule*(
+  delegate: delegate_interface.AccessInterface,
+  events: EventEmitter,
+  walletAccountService: wallet_account_service.Service,
+  networkService: network_service.Service,
+): Module =
+  result = Module()
+  result.delegate = delegate
+  result.events = events
+  result.walletAccountService = walletAccountService
+  result.view = newView(result)
+  result.controller = controller.newController(result, events, walletAccountService)
+  result.moduleLoaded = false
+  result.authentiactionReason = AuthenticationReason.DeleteAccountAuthentication
+
+method delete*(self: Module) =
+  self.view.delete
+  self.controller.delete
+
+method refreshWalletAccounts*(self: Module) =
+  let walletAccounts = self.controller.getWalletAccounts()
+
+  let items = walletAccounts.map(w => (block:
+    walletAccountToItem(w)
+  ))
+
+  self.view.setItems(items)
+
+method load*(self: Module) =
+  singletonInstance.engine.setRootContextProperty("profileSectionAccountsModule", newQVariant(self.view))
```

Comment:
@alaibe I would rather remove this line and expose accounts via `profileSectionModule`, like it's done [here](https://github.com/status-im/status-desktop/blob/master/src/app/modules/main/profile_section/module.nim#L199-L200), that way developers won't be able to use it from wherever in the app, but only from the places where context of `profileSectionModule` is known.